### PR TITLE
Add types for epub-gen

### PR DESCRIPTION
**Pull request by styler3**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `sint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/epub-gen/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
+// Type definitions for epub-gen 0.1
+// Project: https://github.com/cyrilis/epub-gen
+// Definitions by: Sam Tyler <https://github.com/styler3>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export interface Chapter {
+    /** Chapter title */
+    title?: string;
```
Comment (jakebailey) [inline types/epub-gen/index.d.ts:8]: I highly suspect that these should all have `| undefined` to support `exactOptionalPropertyTypes`.

See: [Playground Link](https://www.typescriptlang.org/play?exactOptionalPropertyTypes=true&ts=4.4.2#code/PTAEHUEsBcAtQKYA8CGBjaB5ADtSB7AOxQBsAFAJ32wQugE8AVemgZ0WICMSEATAGlDMaAZTQVIuALAAoEKFgp2raBIy1QFAK492KKlsK9QAd0XRQ9fFtBoUhUKwQX7oaniKlZ87FRp16UGh8ADpZbzAAMXwKRFQAW2weQThIdkhCaFoAM3QEUF4ENBJ9BHY4FAs4Wny00BQI0F9qWgZTWEg0eDsHTnyifuygk3xHVQyAc1YALlAAcl59AGs5t1i5kkgJ2Gg5xpjQGEdYaxJjQnwLPsOHardOACsi6DCZWQysily0fIBVJwoABEELkdNB2ABvWSgUDyRiwfIoThOQg-NxDFCgABupC0+QoCF8ZQQmXYc1Y9BUCHiexkMLQ+BIMXh1IQmCxtAkhQA-LMAESLChLPmgAA+oD5m220D5AG5ZABfcJyMBQODWKoddjZEoTDhIngCIIIgn1U1wAn5HEkPHsMydeBWGw9RpOCwAAwZTIoLPibI5FC5CHdQVGfX5guFgklWx2IvsxndhkK2QyfHdr0aImceEIeru1rxodASaMILTvBDJkgJBI9Vr+BMoHi+BUoAAUigcWIJLhNIY8H7Gl0ikttQc7sg0llUflgqBcjXBPauod2CDss9IBySIFciQKa95AAJRsIAMpE0Icn1BdaABe97TrHKo1S7D6iixBAoISECIKctCBgAhCEabIDk9RlmQRP12U5SBChDAB9Xh8DKQg5mgZCbmKLRChLZMgPTTMVX-fIdRQPV3WQdAsFwUDSEoFoAmEMoQ3iFAljKIQWAQHtJBcNA0C0ChKgQXdGk46BVzuFMMhAogmiob9CmMJR6gcdxGLrZp-AYUjCmKUo71RDwHAmZx-lobNoFzKYAAoAEpZmsoEQRQMFWHlN4ZAZQg2zdez2AAXlASzoDc2zgucnygsmVgQi9GDWXgwNEPyMKBWWOVZHivNEuSn1YP9BCCKyqU4x8xoACEtAsC5oGmPKcwSpLoOK1KAyDUAwqI1NCD4aqyMoH9izuSjqNojAcHMpi-FaJg+NYd1LzqEwNMappW1YSBuAQEIgA)

File: types/epub-gen/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    customNcxTemplatePath?: string;
+    /** For advanced customizations: absolute path to a HTML toc template. */
+    customHtmlTocTemplatePath?: string;
+    /** Book Chapters content. */
+    content: Chapter[];
+    /**  Whether or not to log progress messages */
+    verbose?: boolean;
+}
+
+export default class Epub {
```
Comment (peterblazejewicz) [inline types/epub-gen/index.d.ts:56]: it looks this is CommonJS module exporting class symbol:
```ts
 const Epub = require("epub-gen");
```
https://github.com/cyrilis/epub-gen/blob/master/lib/index.js#L438
(check the readme of the module for usage via CJS import)

Comment (styler3) [reply types/epub-gen/index.d.ts:56]: Thanks @peterblazejewicz.
Could you please let me know what change you need?

File: types/epub-gen/index.d.ts (hunk 1 above)
Comment (styler3) [reply types/epub-gen/index.d.ts]: Thanks.
Yeah, I tested it. You're right. I have updated.

File: types/epub-gen/index.d.ts (hunk 2 above)
Comment (jakebailey) [reply types/epub-gen/index.d.ts:56]: You would need to write something like:

```ts
class Epub { }

export = Epub
```

There's an example in the README: https://github.com/DefinitelyTyped/DefinitelyTyped#esmoduleinteropallowsyntheticdefaultimports

Comment (styler3) [reply types/epub-gen/index.d.ts]: Thanks for your help :+1:.
I have pushed an update.

File: types/epub-gen/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
+// Type definitions for epub-gen 0.1
+// Project: https://github.com/cyrilis/epub-gen
+// Definitions by: Sam Tyler <https://github.com/styler3>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+interface Chapter {
```
Comment (jakebailey) [inline types/epub-gen/index.d.ts:6]: I _believe_ types won't be available on the export anymore, now that you've done `export = Epub`.

You likely (?) want to combine this with a `namespace` to merge onto `Epub` so you can reference it. I think there are some examples of this in the repo if you do a little searching.

Comment (styler3) [reply types/epub-gen/index.d.ts]: Sorry @jakebailey, I've never worked with either CommonJS nor used `namespace` before so I don't really understand what I'm doing here. It did seem to work for me the way it was before :shrug:.

I've updated it basing it on `types/11ty__eleventy-img/generate-html.d.ts` as an example.

Is that right?

Comment (jakebailey) [reply types/epub-gen/index.d.ts]: I think what you've done looks correct now, yeah.

**Review by jakebailey — APPROVED** [review]

**Comment by styler3** [comment]
Ready to merge

**Pull request merged**